DEPENDENCY_POLICY.md

```diff
@@ -0,0 +1,102 @@
+# Envoy External Dependency Policy
+
+Envoy has an evolving policy on external dependencies, tracked at
+https://github.com/envoyproxy/envoy/issues/10471. This will become stricter over time, below we
+detail the policy as it currently applies.
+
+## External dependencies dashboard
+
+The list of external dependencies in Envoy with their current version is available at
+https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/external_deps
+
+## Declaring external dependencies
+
+In general, all external dependencies for the Envoy proxy binary build and test should be declared
+in either [bazel/repository_locations.bzl](bazel/repository_locations.bzl) or
+[api/bazel/repository_locations.bzl](api/bazel/repository_locations.bzl), unless listed under
+[policy exceptions](#policy-exceptions).
+
+An example entry for the `nghttp2` dependency is:
+
+```python
+com_github_nghttp2_nghttp2 = dict(
+    project_name = "Nghttp2",
+    project_url = "https://nghttp2.org",
+    version = "1.41.0",
+    sha256 = "eacc6f0f8543583ecd659faf0a3f906ed03826f1d4157b536b4b385fe47c5bb8",
+    strip_prefix = "nghttp2-{version}",
+    urls = ["https://github.com/nghttp2/nghttp2/releases/download/v{version}/nghttp2-{version}.tar.gz"],
+    use_category = ["dataplane"],
+    cpe = "cpe:2.3:a:nghttp2:nghttp2:*",
+),
+```
+
+Dependency declarations must:
+
+* Provide a meaningful project name and URL.
+* State the version in the `version` field. String interpolation should be used in `strip_prefix`
+  and `urls` to reference the version. If you need to reference version `X.Y.Z` as `X_Y_Z`, this
+  may appear in a string as `{underscore_version}`, similarly for `X-Y-Z` you can use
+  `{dash_version}`.
+* Versions should prefer release versions over master branch GitHub SHA tarballs. A comment is
+  necessary if the latter is used.
```

Comment:
Can you detail the comment format? This comes up pretty often (I assume date imported, reason(s) not using a release version, etc.). cc @moderation

Reply:
Added. I actually plan on introducing a `last_updated` metadata field later, but will leave in the comment for now.

Reply:
@htuch 's last update clarifying date format looks good. To really get some standardization I'd dictate that the date comment comes on the line after `project_url`. An example of non-conforming is https://github.com/envoyproxy/envoy/blob/master/bazel/repository_locations.bzl#L123. As soon as my pip3 PR merges I have a dataplane PR and will clean these up.

I like the idea of `last_updated` metadata (but probably only for the non-release deps)

Reply:
For standardizing and consistency, I recommend holding up until we fix the last updated metadata, since that's the long term direction for sure.